sys/sched_round_robin/sched_round_robin.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2021 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       Round Robin Scheduler implementation
+ *
+ * @author      Karl Fessel <karl.fessel@ovgu.de>
+ *
+ * @}
+ */
+
+#include "sched.h"
+#include "thread.h"
+#include "xtimer.h"
+#include "sched_round_robin.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static void sched_round_robin_cb(void * d);
+
+static xtimer_t rr_timer = {.callback = sched_round_robin_cb};
+
+static uint8_t current_rr_priority;
```

Comment:
Do we need C11 atomics or `sys/atomic_utils` here? I haven't heard of any MCU being able to get an inconsistent read from byte yet - this would require memory bus with less than 8 bits I guess. But I guess adding a comment like

```c
/* assuming simple reads from and writes to a byte to be atomic on every board */
```

doesn't hurt. Because the very moment someone starts doing read-modify-write operations on this, those wouldn't be atomic anymore.